### PR TITLE
Added fastq_bwa_mem_samblaster

### LINKED ISSUE #4909: new subworkflow: FASTQ_BWA_MEM_SAMBLASTER
### Is there an existing subworkflow for this?

- [X] I have searched for the existing subworkflow

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this subworkflow, I added myself to the `Assignees` to facilitate tracking who is working on the subworkflow

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #4909 

- [x] Added fastq_bwa_mem_samblaster
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by SPPearce** [comment]
Do you know if samblaster works on CRAM files?

**Comment by GallVp** [comment]
> Do you know if samblaster works on CRAM files?

According to this the developer has no plans to support CRAM for now: https://github.com/GregoryFaust/samblaster/issues/25#issuecomment-257062937

File: subworkflows/nf-core/fastq_bwa_mem_samblaster/tests/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+process {
+    withName: BWA_MEM {
+        ext.prefix = { "${meta.id}.on.${meta.ref_id}.bwa.mem" }
```
Comment (SPPearce) [inline subworkflows/nf-core/fastq_bwa_mem_samblaster/tests/nextflow.config:3]: Why these prefixes?

**Review by SPPearce — COMMENTED** [review]
Can we swap to the . notation that the other subworkflows generally use?

**Comment by GallVp** [comment]
> Can we swap to the . notation that the other subworkflows generally use?

Thank you for the review.

Do you mean swapping `|` with `.`? Yes, I have no objection to that other than that I prefer `|` as it is visually more explicit compared to a small `.`

File: subworkflows/nf-core/fastq_bwa_mem_samblaster/tests/nextflow.config (hunk 1 above)
Comment (GallVp) [reply subworkflows/nf-core/fastq_bwa_mem_samblaster/tests/nextflow.config:3]: Although these prefixes are not necessary here but in a pipeline they are as fastqs from each sample are mapped to each reference.

**Comment by GallVp** [comment]
Closing this as it has not gained any interest from any other pipeline. If someone comes looking for this or something similar, please find it here: https://github.com/GallVp/nxf-components/blob/c30c8320078c68e14e41516ec6eaaca5f084ef38/subworkflows/gallvp/fastq_bwa_mem_samblaster/main.nf#L5

**Pull request closed without merging**